python/cuml/tests/experimental/test_filex.py

```diff
@@ -265,9 +277,13 @@ def test_fil_regression(
         fil_preds = np.asarray(fm.predict(X_validation))
         fil_preds = np.reshape(fil_preds, np.shape(xgb_preds))
         fil_mse = mean_squared_error(y_validation, fil_preds)
+        fm.optimize(data=X_validation)
+        fil_preds_opt = np.asarray(fm.predict(X_validation))
+        fil_preds_opt = np.reshape(fil_preds_opt, np.shape(xgb_preds))
 
         assert fil_mse == pytest.approx(xgb_mse, abs=0.01)
         assert np.allclose(fil_preds, xgb_preds, 1e-3)
+        assert np.allclose(fil_preds_opt, fil_preds, 1e-3)
```

Comment:
I would compare both outputs to the same value, otherwise the tolerance on the optimized preds would be much higher.
```suggestion
        assert np.allclose(fil_preds_opt, xgb_preds, 1e-3)
```
Alternatively, you can reduce the tolerance between `fil_preds_opt` and `fil_preds`.

Reply:
With different chunk sizes, you actually can get differences in the FIL output of the same order as the difference with the XGBoost prediction. In both cases, the problem is that you get different floating point drift due to differences in the order of parallel summations. I prefer the current version because it indicates the intent of the test a bit better (ensuring that optimization does not substantively change the output from what we get without optimization), but I could be swayed.

Reply:
But isn't the ultimate goal to ensure that the FIL predictions are within a specific tolerance of our ground truth, in this case the XGBoost predictions? It shouldn't matter whether we use the optimized or non-optimized parameters, shouldn't it? As a user I would be concerned if the accuracy potentially suffers using the optimization.

If you want to ensure that the optimized results are within a specific tolerance of the non-optimized FIL results, then I would expect the tolerance to be lower.